tests/models/whisper/test_modeling_whisper.py

```diff
@@ -0,0 +1,851 @@
+        ).input_features.to(torch_device)
+        labels = processor(audio=input_speech, text="This part of the speech", return_tensors="pt").labels.to(
+            torch_device
+        )
+        with torch.no_grad():
+            logits = model(
+                input_features,
+                decoder_input_ids=labels,
+                output_hidden_states=False,
+                output_attentions=False,
+                use_cache=False,
+            )
+
+        logits = logits.last_hidden_state @ model.decoder.embed_tokens.weight.T
+
+        # fmt: off
+        EXPECTED_LOGITS = torch.tensor(
+            [
+                8.8958,  4.0423,  9.8841,  9.8493, 10.0628,  4.8472,  9.0100,  5.7364,
+                5.9165,  7.6322,  3.1579, 10.7269,  6.9586, 10.1852,  5.4714,  8.2995,
+                4.7507,  6.6723,  7.2764,  7.1831,  7.0388,  7.2191,  6.2364,  6.2117,
+                5.8797,  2.8099,  6.8319,  5.7094,  0.6999,  6.8444
+            ]
+        )
+        # fmt: on
+        self.assertTrue(torch.allclose(logits[0, 0, :30].cpu(), EXPECTED_LOGITS, atol=1e-4))
+
+    def test_large_logits_librispeech(self):
+
+        from transformers import GPT2Tokenizer, WhisperFeatureExtractor, set_seed
+
+        torch_device = "cpu"
+        set_seed(0)
+        model = WhisperModel.from_pretrained("whisper/large")
+        model.to(torch_device)
+
+        # processor = self.default_processor
+
+        input_speech = self._load_datasamples(1)
+
+        feaure_extractor = WhisperFeatureExtractor()
+        tokenizer = GPT2Tokenizer.from_pretrained("gpt2")
+        tokenizer.pad_token = 0
+
+        processor = WhisperProcessor(feaure_extractor, tokenizer)
+
+        input_features = processor(
+            audio=input_speech, text="This part of the speech", return_tensors="pt"
+        ).input_features.to(torch_device)
+        labels = processor(audio=input_speech, text="This part of the speech", return_tensors="pt").labels.to(
+            torch_device
+        )
+        with torch.no_grad():
+            logits = model(
+                input_features,
+                decoder_input_ids=labels,
+                output_hidden_states=False,
+                output_attentions=False,
+                use_cache=False,
+            )
+
+        logits = logits.last_hidden_state @ model.decoder.embed_tokens.weight.T
+
+        # fmt: off
+        EXPECTED_LOGITS = torch.tensor(
+            [
+                2.1807,  1.1505,  4.8049,  3.9549,  2.7182,  4.1885, -0.4179,  2.8316,
+                2.0155,  2.2740,  2.6727,  1.3789,  0.5620,  2.2096,  1.6781,  2.8227,
+                1.4421,  0.9057,  1.3358,  2.2104,  2.7468,  2.0021,  2.6960,  1.5925,
+                2.2239,  1.9396,  4.0580,  5.7722,  4.8056,  4.2416
+            ]
+        )
+        # fmt: on
+
+        self.assertTrue(torch.allclose(logits[0, 0, :30].cpu(), EXPECTED_LOGITS, atol=1e-4))
+
+    def test_generationt(self):
```

Comment:
think at least we should add 3,4 tests here 2 and 2 for multi-lingual and english-only

Reply:
On it !